iceoryx_utils/include/iceoryx_utils/internal/concurrent/periodic_task.hpp

```diff
@@ -51,15 +58,32 @@ class PeriodicTask
     PeriodicTask& operator=(const PeriodicTask&) = delete;
     PeriodicTask& operator=(PeriodicTask&&) = delete;
 
+    /// @brief Spawns a thread and immediately executes the callable specified with the constructor.
+    /// The execution is repeated after the specified interval is passed.
+    /// @param[in] interval is the time the thread waits between two invocations of the callable
+    /// @attention If there is already a running thread, this will be stopped and started again with the new interval.
```

Comment:
You mean a running thread with the same name right?

Reply:
no, the `PeriodicTask` manages one thread. If that thread is running, it is stopped. Maybe it's a bit misleading from the text